src/infrastructure/Models/SearchCriteria.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Infrastructure\Models;
+
+class SearchCriteria
```

Comment:
I don't see any API how build your criteria. It's full black box and doesn't look usable.

Reply:
Maybe it can be a class or at list somethig how build it and what it accepts

Reply:
It parses regular URL query string. Documentation will be added to the swagger schema.
Look into internal class creation for service usage